src/Controller/ShopwareFilesController.php

```diff
@@ -175,4 +189,22 @@ private function isIgnoredFileHash(string $file): int
 
         return self::STATUS_OK;
     }
+
+    private function getUserName(Context $context): ?string
+    {
+        $contextSource = $context->getSource();
+
+        if (!$contextSource instanceof AdminApiSource) {
+            return null;
+        }
+
+        /** @var null|UserEntity $userEntity */
+        $userEntity = $this->userRepository->search(new Criteria([$contextSource->getUserId()]), Context::createDefaultContext())->first();
```

Comment:
getUserId can be null when integrationId is filled

Reply:
Thank you! Integration is now supported, too.